src/application/common/logger.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as Logger from 'bunyan';
+
+const ILogger = Symbol("ILogger");
+
+interface ILogger {
+    addStream(stream: Logger.Stream): Promise<void>;
```

Comment:
It would be nice to have something simple done first and then look at other features and add them if we have a need for them.

1. child sounds good
2. if we provide setters for levels we also need the client interface to receive notifications about it:
```ts
export interface ILoggerClient {
  onLevelChanged(event: {
     loggerId: string; // some kind of and id to identify a responsible logger in case if we support children
     oldLogLevel: number;
     newLogLevel: number;
  }): void;
}
```

And ILogger can look like:
```ts
export interface ILogger {
    readonly id: string;
    error, info, warn ...
    createChild(id: string): ILogger;
    readonly onLevelChanged: Event<{oldLogLevel: number, newLogLevel: number}>
}
```
3. logging functions: we can have params, for now it will be up to users to provide proper JSON objects. Later it would be nice to add `serializers` support generally to JSON-RPC proxies, so we can use any type, like `URI` in `FileSystem` interface. The same serializers mechanics can be used to customize serialization/parsing logging params. We should do it separately from this PR.
4. there are attempts provide streaming over JSON-RPC: https://github.com/Microsoft/language-server-protocol/pull/182. We should look at it also separately.